Tools/scripts/duplicate_meth_defs.py

```diff
@@ -0,0 +1,271 @@
+#!/usr/bin/env python3
+"""Print duplicate method definitions in the same scope."""
+
+import sys
+import os
+import ast
+import re
+import argparse
+import traceback
+from tokenize import open as tokenize_open
+from contextlib import contextmanager
+
+class ClassFunctionVisitor(ast.NodeVisitor):
+    def __init__(self, duplicates):
+        self.duplicates = duplicates
+        self.scopes = []
+
+    @contextmanager
+    def _visit_new_scope(self, node):
+        prev_scope = self.scopes[-1] if len(self.scopes) else None
+        new_scope = Scope(node, prev_scope)
+        try:
+            yield prev_scope, new_scope
+        finally:
+            self.scopes.append(new_scope)
+            self.generic_visit(node)
+            self.scopes.pop()
+
+    def visit_FunctionDef(self, node):
+        with self._visit_new_scope(node) as (previous, new):
+            if previous is not None and previous.type == 'ClassDef':
+                new.check_duplicate_meth(self.duplicates)
+
+    def visit_ClassDef(self, node):
+        with self._visit_new_scope(node) as (previous, new):
+            pass
+
+    visit_AsyncFunctionDef = visit_ClassDef
```

Comment:
Should this alias to `visit_FunctionDef` ? This doesn't catch below case and the pattern is seen in `Lib/test/test_sys_settrace.py` where async functions are defined inside a class. E.g. https://github.com/python/cpython/blob/3e986de0d65e78901b55d4e500b1d05c847b6d5e/Lib/test/test_sys_settrace.py#L737

```python
class A:

    async def foo(self):
        pass

    async def foo(self):
        pass
```

Reply:
Right, thanks.
Fixed in commit ea41f060b776230b31ef24ddf3539661d2886937.